pkg/upgrade/migrate_deployment_config_2.15.go

```diff
@@ -0,0 +1,55 @@
+package upgrade
+
+import (
+	"context"
+	"fmt"
+	"github.com/3scale/3scale-operator/pkg/helper"
+	appsv1 "github.com/openshift/api/apps/v1"
+	k8sappsv1 "k8s.io/api/apps/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// MigrateDeploymentConfigToDeployment verifies the Deployment is healthy and then deletes the corresponding DeploymentConfig
+// 3scale 2.14 -> 2.15
+func MigrateDeploymentConfigToDeployment(dName string, dNamespace string, client k8sclient.Client) (bool, error) {
+	deploymentConfig := &appsv1.DeploymentConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      dName,
+			Namespace: dNamespace,
+		},
+	}
+	err := client.Get(context.TODO(), k8sclient.ObjectKey{Name: deploymentConfig.Name, Namespace: deploymentConfig.Namespace}, deploymentConfig)
+
+	// Breakout if the DeploymentConfig has already been deleted
+	if k8serr.IsNotFound(err) {
+		return true, nil
+	}
+	if err != nil && !k8serr.IsNotFound(err) {
```

Comment:
nitpicking here
```go
if err != nil  {
    ...
}
```
should be enough, right?

Reply:
Good catch, will change